src/components/structures/BackdropPanel.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2021 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { createRef } from "react";
+import "context-filter-polyfill";
+
+interface IProps {
+    width?: number;
+    height?: number;
+    backgroundImage?: CanvasImageSource;
+    blur?: string;
+}
+
+
+export default class BackdropPanel extends React.PureComponent<IProps> {
```

Comment:
is there any major performance difference in keeping the canvas in DOM vs exporting to data uri and then using css background?

Reply:
I believe this right way to achieve that would be to use Houdini and the [Paint API](https://developer.mozilla.org/en-US/docs/Web/CSS/paint()) 
Unfortunately browser support is not quite there yet

I don't think that removing the canvas from the DOM would bring major performance improvements, and `toDataURL` is a synchronous operation that can take a long time, all of that to do another painting operation that already happened on the canvas itself

Let me know if you can think about any pitfalls with the current approach

Reply:
capturing the bg image as toDataURL sounds very likely to stall things as it passes around the big data URL (e.g. the horrible perf problems we had on FF back when E2EE images were expressed as data URLs rather than blobs)